scripts/quick-deploy-service.ts

```diff
@@ -0,0 +1,393 @@
+  accessToken: string,
+  serviceId: string,
+  dockerUsername: string,
+): Promise<void> => {
+  const env = from(process.env);
+
+  const hostingServiceBaseUrl = env
+    .get('HOSTING_SERVICE_BASE_URL')
+    .required()
+    .asString();
+
+  const createServiceDefinitionResponse = await axios.post(
+    new URL('graphql', hostingServiceBaseUrl).href,
+    {
+      query: print(gql`
+        mutation CreateServiceDefinition(
+          $input: CreateServiceDefinitionInput!
+        ) {
+          createServiceDefinition(input: $input) {
+            serviceDefinition {
+              id
+            }
+          }
+        }
+      `),
+      variables: {
+        input: {
+          serviceDefinition: {
+            serviceId: serviceId,
+            name: serviceId,
+            repositoryName: `${dockerUsername}/${serviceId}`,
+          },
+        },
+      },
+    },
+    {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    },
+  );
+
+  if (
+    createServiceDefinitionResponse.data.errors &&
+    createServiceDefinitionResponse.data.errors[0].message !==
+      'Attempt to create or update an element failed, as it would have resulted in a duplicate element.'
+  ) {
+    throw new Error(createServiceDefinitionResponse.data.errors[0].message);
+  }
+};
+
+/**
+ * Queries the service definition ID for the given service ID.
+ */
+const getServiceDefinitionID = async (
+  accessToken: string,
+  serviceId: string,
+): Promise<string> => {
+  const env = from(process.env);
+
+  const hostingServiceBaseUrl = env
+    .get('HOSTING_SERVICE_BASE_URL')
+    .required()
+    .asString();
+
+  const getServiceDefinitionResponse = await axios.post(
+    new URL('graphql', hostingServiceBaseUrl).href,
+    {
+      query: print(gql`
+        query GetServiceDefinition($condition: ServiceDefinitionCondition!) {
+          serviceDefinitions(condition: $condition) {
+            nodes {
+              id
+              name
+            }
+          }
+        }
+      `),
+      variables: {
+        condition: {
+          serviceId: serviceId,
+        },
+      },
+    },
+    {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    },
+  );
+
+  if (getServiceDefinitionResponse.data.errors) {
+    throw new Error(getServiceDefinitionResponse.data.errors[0].message);
+  }
+
+  console.log(
+    chalk.cyan(
+      `\nService Definition for [${serviceId}] found in the environment. Continuing with the deployment.`,
+    ),
+  );
+
+  return getServiceDefinitionResponse.data.data.serviceDefinitions.nodes[0].id;
+};
+
+/**
+ * Validate if the deployment manifest YAML file of the service still has some variables that need to be manually adjusted.
+ */
+function validateDeploymentManifestIsModified(
+  serviceId: 'media-service' | 'catalog-service' | 'entitlement-service',
+): void {
+  const servicePrefix = serviceId.split('-service')[0];
+
+  if (serviceId === 'entitlement-service') {
```

Comment:
Any reason why we're only doing this for `entitlement-service` but accept arguments for all other services as well? Feel like the validation below, of checking for `REPLACE_WITH_VALID_` is generic enough for all services?

Reply:
no actually - this was probably a remnant filter condition I had while evolving the script. it should be generic enough to work for all services given they all follow the same naming convention for client replicable variables.

thanks!